ee/server/service/teams.go

```diff
@@ -499,9 +500,21 @@ func (svc *Service) teamByIDOrName(ctx context.Context, id *uint, name *string)
 
 var jsonNull = json.RawMessage(`null`)
 
+// setAuthCheckedOnPreAuthErr can be used to set the authentication as checked
+// in case of errors that happened before an auth check can be performed.
+// Otherwise the endpoints return a "authentication skipped" error instead of
+// the actual returned error.
+func setAuthCheckedOnPreAuthErr(ctx context.Context) {
+	if az, ok := authz_ctx.FromContext(ctx); ok {
+		az.SetChecked()
+	}
+}
+
 func (svc *Service) ApplyTeamSpecs(ctx context.Context, specs []*fleet.TeamSpec, applyOpts fleet.ApplySpecOptions) error {
-	if err := svc.authz.Authorize(ctx, &fleet.Team{}, fleet.ActionRead); err != nil {
-		return err
+	if len(specs) == 0 {
+		setAuthCheckedOnPreAuthErr(ctx)
```

Comment:
Should this also be called here? https://github.com/fleetdm/fleet/pull/10850/files#diff-967da913375c2a1d5b2edb39e5e37ff5a41b8a5b73569a73a62b12c3330da627R531

Reply:
Sorry, the link links to the whole diff

Reply:
Sorry - https://github.com/fleetdm/fleet/blob/aa4cd1f201ed8d133de37431df656ac0de547704/ee/server/service/teams.go#L531

Reply:
Good idea. Done! https://github.com/fleetdm/fleet/pull/10850/commits/64f6a4a5398b881676309c7e1319b5cbcbf7fbfb